pkg/controller/certificates/groupapprove.go

```diff
@@ -72,7 +72,7 @@ func (cc *groupApprover) AutoApprove(csr *certificates.CertificateSigningRequest
 	if len(x509cr.DNSNames)+len(x509cr.EmailAddresses)+len(x509cr.IPAddresses) != 0 {
 		return csr, nil
 	}
-	if !hasExactUsages(csr, kubeletClientUsages) {
+	if !hasExactUsages(csr, kubeletClientUsages) && !hasExactUsages(csr, kubeletServerUsages) {
```

Comment:
Agree. There are three distinct autoapproval cases we want to detect (see https://github.com/kubernetes/kubernetes/issues/45030)

1. CSR shaped like a kubelet client cert, requested by the kubelet itself
2. CSR shaped like a kubelet serving cert, requested by the kubelet itself
2. CSR shaped like a kubelet client cert, requested by a different user (like a shared bootstrap user)